### ISSUE TITLE
Armor not Charging

### ISSUE THREAD

**Issue by IAmTheShark223** [MachineMuse/MachineMusePowersuits#883]
I'm in the latest alpha for 1.12.2 for powersuits, a few of the modules like the night vision and all generators are not working. i am using ALPHA-23,

**Comment by arnesacnussem** [comment]
have you installed a battery?

**Comment by IAmTheShark223** [comment]
> have you installed a battery?

yes, I have lv batteries on all my pieces

**Comment by IAmTheShark223** [comment]
as an additional note: the generators and modules work in singleplayer but not on my server. i have no clue what is causing this

**Comment by lehjr** [comment]
Sounds like a sync issue between client and server. It probably is charging on the server since all energy is handled server side, but not showing as such on the client.

**Comment by IAmTheShark223** [comment]
that isnt the issue; at least i dont think it is, stuff like the night vision module and the generators just don't work on the server, nighvision doesnt apply the effect in multiplayer but does in singleplayer. i think it might be a client mod that's running on the server but i have no way of telling a client mod apart from a server mod.

**Comment by IAmTheShark223** [comment]
since you need the modlist to help me with that, here it is. 


![server mod list](https://user-images.githubusercontent.com/46060978/50379424-4ccf5900-05fe-11e9-9e1c-a7b22013b8cb.jpg)

**Comment by lehjr** [comment]
Yeah, looks like I'll have to do some trouble shooting.

**Comment by IAmTheShark223** [comment]
I removed one mod i thought would be the issue, it's called defaultoptions or something like that

**Comment by lehjr** [comment]
From what I can tell, it's likely just client<->server sync issues. I'll have to dig into it a bit more though.

**Comment by IAmTheShark223** [comment]
I'll put up whatever information is necessary, i can give you a copy of the server and my modpack is on curse

**Comment by IAmTheShark223** [comment]
How would i be able to check for a sync issue?

**Comment by lehjr** [comment]
Sync issue is a bug in the mod. Already looking into it. So far found one issue.

**Comment by IAmTheShark223** [comment]
Ok

**Comment by eyeonus** [comment]
If you go into the server's 'mods' folder, open a console/terminal, and type either 'dir >modlist.txt' on Win or 'ls >modlist.txt' on *nix/Mac, you'll get a text file that lists all your mods, which might be slightly easier to use/get than a screenshot.

I could've sworn there was a website online that would spit out a modlist for the purpose of sharing with other people, but I can't find it.

**Comment by IAmTheShark223** [comment]
Thanks, I think lehjr narrowed the issue down to the mod itself but i'll do that.

**Comment by lehjr** [comment]
well, one major issue is the PlayerUpdateHandler (event handler) being registered in the client proxy instead of common proxy. This has a few functions on both client and server, and one important one being the tick handler for the modules.

**Comment by IAmTheShark223** [comment]
Ah, so in layman's terms, the mod is telling the client to do things but forgets that the server is there?

**Comment by lehjr** [comment]
Yeah, for the most part. Some code is set up so it only runs on the server side, so that gets skipped altogether. Anyway, I'll have a fixed version up in the morning.

**Comment by IAmTheShark223** [comment]
Thank you so much

**Comment by lehjr** [comment]
Uploaded, should be visible ... "shortly"?

**Comment by IAmTheShark223** [comment]
Thank you so much
Will test it as soon as i can

**Comment by IAmTheShark223** [comment]
It works. Thank you so much for doing this.

**Comment by IAmTheShark223** [comment]
one of my testers for my pack just noted that the slider for the kinetic generator is not working.

**Comment by lehjr** [comment]
Yeah, looks like NONE of the sliders are working.

**Comment by lehjr** [comment]
fixed.

**Comment by IAmTheShark223** [comment]
Thanks.

**Comment by eyeonus** [comment]
It was more of a for future reference kind of thing.
